templates/_worker.js/handleRequest.ts

```diff
@@ -76,12 +77,16 @@ async function generateResponse(
 		return new Response(null, { status, headers: headers.normal });
 	}
 
-	let resp = await runOrFetchBuildOutputItem(output[path], reqCtx, {
-		path,
-		status,
-		headers,
-		searchParams,
-	});
+	let resp =
+		body !== undefined
+			? // If we have a response body from matching, use it instead.
+			  new Response(body, { status })
```

Comment:
This probably isn't the *best* way to deal with a body existing, but I guess it'll do for now, at least until this function eventually gets refactored

Reply:
why do you think it isn't the best way to deal with it? what would be? is there a need to refactor this function?

Reply:
> why do you think it isn't the best way to deal with it? what would be? is there a need to refactor this function?

because it would now be unnecessarily creating the response twice in this case. here and then just a little bit below.

Reply:
ah yeah... I see 🤔